src/compas_fab/robots/robot.py

```diff
@@ -744,14 +764,26 @@ def to_world_coordinates(self, frame_RCF, group=None):
         frame_WCF = frame_RCF.transformed(self.transformation_RCF_WCF(group))
         return frame_WCF
 
-    def from_tcf_to_t0cf(self, frames_tcf):
+    def _get_attached_tool_for_group(self, group_name=None):
+        """Get the tool attached to the given planning group. Group name defaults to main_group_name.
+        Raises ValueError if group name is unknown or there is no tool currently attached to it"""
+        group = group_name or self.main_group_name
+        if not (group in self.attached_tools and self.attached_tools[group]):
+            raise ValueError("Planning group {} not found or no tool attached".format(group))
```

Comment:
mmh it turns into an `or` then, meaning two subsequent `if`s with the same `raise` statement. But I agree readability here is not great.

How is this instead?
```python
    def _get_attached_tool_for_group(self, group_name=None):
        group = group_name or self.main_group_name
        if group in self.attached_tools and self.attached_tools[group]:
            return self.attached_tools[group]

        raise ValueError("Planning group {} not found or no tool attached".format(group))
```

Reply:
but since the `raise` message also has an "or", isn't it possible to split the error messages as well?
```python

def _get_attached_tool_for_group(self, group_name=None):
    group = group_name or self.main_group_name
    if not group in self.attached_tools:
        raise ValueError("Planning group {} not found".format(group))

    if not self.attached_tools[group]:
        raise ValueError("No tool attached")

    return self.attached_tools[group]
````

Reply:
Makes sense! 👍